### PR TITLE
Add StringArrayConverter to handle string arrays in Gallery-style config

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/Engineering/issues/3020

We have some internal configuration that we want to pull into the private build of gallery that has an array configuration variable. The way we do configuration in gallery is via `TypeConverter`/component model stuff and our config comes from string to string pairs. Therefore if the configuration model has a string array in it, this is how we handle it.

Note that we already have the implementation here in gallery:
https://github.com/NuGet/NuGetGallery/blob/master/src/NuGetGallery.Services/Helpers/StringArrayConverter.cs

However I'd like to move it to a common place so that an internal project that gallery will depend on with build magic can use it too.

### CODE REVIEW

File: tests/NuGet.Services.Configuration.Tests/StringArrayConverterFacts.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            var target = new StringArrayConverter();
+
+            var output = target.ConvertFrom(input);
+
+            var array = Assert.IsType<string[]>(output);
+            Assert.Equal(new[] { input }, array);
+        }
+
+        [Fact]
+        public void ReturnsEmptyArrayWithEmpty()
```
Comment (loic-sharma) [inline tests/NuGet.Services.Configuration.Tests/StringArrayConverterFacts.cs]: Should we also have a test case for `null` values?

Comment (joelverhagen) [reply tests/NuGet.Services.Configuration.Tests/StringArrayConverterFacts.cs]: `null` should never happen per the type converter contract. The `CanConvertFrom` will return `null` and the method will never be called.
